core/src/main/java/nl/nn/adapterframework/doc/model/FrankElement.java

```diff
@@ -0,0 +1,249 @@
+package nl.nn.adapterframework.doc.model;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.Logger;
+import org.springframework.core.annotation.AnnotationUtils;
+
+import lombok.Getter;
+import lombok.Setter;
+import nl.nn.adapterframework.doc.IbisDoc;
+import nl.nn.adapterframework.doc.IbisDocRef;
+import nl.nn.adapterframework.util.LogUtil;
+
+public class FrankElement {
+	private static Logger log = LogUtil.getLogger(FrankElement.class);
+
+	private static final String JAVA_STRING = "java.lang.String";
+
+	private final @Getter String fullName;
+	private final @Getter String simpleName;
+	private @Getter FrankElement parent;
+	private @Getter List<FrankAttribute> attributes;
+
+	public interface FrankElementStore {
+		boolean hasFrankElement(String name);
+		FrankElement getFrankElement(String name);
+		void addFrankElement(FrankElement frankElement);
+		int numFrankElements();
+	}
+
+	public static FrankElement frankElement(Class<?> clazz, FrankElementStore store) {
+		if(store.hasFrankElement(clazz.getName())) {
+			return store.getFrankElement(clazz.getName());
+		}
+		Class<?> superClass = clazz.getSuperclass();
+		FrankElement parent = superClass == null ? null : frankElement(superClass, store);
+		FrankElement current = new FrankElement(clazz, parent, store);
+		store.addFrankElement(current);
+		return current;
+	}
+
+	FrankElement(Class<?> clazz, FrankElement parent, FrankElementStore store) {
+		this(clazz.getName(), clazz.getSimpleName());
+		this.parent = parent;
+		this.attributes = createAttributes(clazz.getDeclaredMethods(), store);
+	}
+
+	FrankElement(final String fullName, final String simpleName) {
```

Comment:
Heb je deze constructor zonder parent nu al nodig?

Reply:
Ik gebruik hem in de tests. Heb je gezien dat hij niet public is?